doc/changes/10535.md

```diff
@@ -0,0 +1,2 @@
+- Use Merlin new configuration directives `BH` and `SH` for hidden dependencies.
```

Comment:
I would mention that hidden dependencies correspond to the `-h` flag instead of internal merlin directives which most people are unaware of.

Reply:
Indeed, it is always better to write Changes entries from the point of view of the user, not the developer. In this case, something like
```
- Make Merlin/OCaml-LSP aware of "hidden" dependencies used by
  `(implicit_transitive_deps false)` via the `-H` compiler flag.
```
would be more useful for the occasional reader.

Reply:
I took the liberty of tweaking the Changes entry in this manner. Planning to merge once CI passes. Thanks!